### ISSUE TITLE
Multi-character delimiter and quoted fields

### ISSUE THREAD

**Issue by janisdd** [mholt/PapaParse#866]
Recently a user reported an issue to me that multi-character delimiter does not work with quoted fields, e.g.

```
a, b, "c, e", d
```
delimiter: `, ` (comma and a single whitespace)

I tried this on the current demo version (on the website): 
```js
Papa.parse('a, b, "c, e", d', {delimiter: ', '})
```
which gives:
```js
'a', 'b', 'c, e", d'
```
expected

```js
'a', 'b', 'c, e', 'd'
```

I also noticed that in the documentation there is always written `delimiting character` but actually multi-character strings are supported (from what I can see in the code and tests).

I think the issue is somewhere around this line: https://github.com/mholt/PapaParse/blob/master/papaparse.js#L1558

```diff
-if (input[quoteSearch + 1 + spacesBetweenQuoteAndDelimiter] === delim)
+if (input.substr(quoteSearch + 1 + spacesBetweenQuoteAndDelimiter, delimLen) === delim)
```

A test could look like this:

```js
{
  description: "Multi-character delimiter (length 2) with quoted field",
  input: 'a, b, "c, e", d',
  config: { delimiter: ", " },
  notes: "The quotes must be immediately adjacent to the delimiter to indicate a quoted field",
  expected: {
    data: [['a', 'b', 'c, e', 'd']],
    errors: []
  }
}
```

**Comment by christopherfanning** [comment]
I want this too.  Were you able to get this working?

**Comment by janisdd** [comment]
With the proposed changes it works as it should.

**Comment by pokoli** [comment]
Could you please fill a PR with the tests and the proposed change? 

I will be happy to merge it if that solves the problem.

**Comment by janisdd** [comment]
I think the documentation needs to be changed as well. Maybe I could get to this this weekend and create a PR.

**Comment by pokoli** [comment]
@janisdd, Yes a documentation update will be good also. 
Waiting for your PR!

**Comment by pokoli** [comment]
Should be fixed with #879
